koku/masu/processor/tasks.py

```diff
@@ -700,3 +701,23 @@ def convert_to_parquet(
         msg = f"Failed to convert the following files to parquet:{','.join(failed_conversion)}."
         LOG.warn(log_json(request_id, msg, context))
         return
+
+
+@app.task(name="masu.processor.tasks.remove_stale_tenants", queue_name="remove_stale_tenants")
+def remove_stale_tenants(self):
+    """ Remove stale tenants from the tenant api """
+    table_sql = """
+    SELECT schema_name
+      FROM api_customer c
+      LEFT
+      JOIN api_provider p
+        ON c.id = p.customer_id
+      LEFT
+      JOIN api_sources s
+        ON p.uuid::text = s.koku_uuid
+     WHERE s.source_id IS null AND c.date_created < now() - INTERVAL '2 weeks';
+        """
+    with connection.cursor() as cursor:
+        cursor.execute(table_sql)
+        data = cursor.fetchall()
+        Tenant.objects.filter(schema_name__in=[i[0] for i in data]).delete()
```

Comment:
Can we log that an attempt to remove tenants was done and also which ones were removed?

Reply:
removed self